ProcessLib/ProcessVariable.h

```diff
@@ -56,10 +57,21 @@ class ProcessVariable
 	/// Returns a mesh on which the process variable is defined.
 	MeshLib::Mesh const& getMesh() const;
 
-	void initializeDirichletBCs(MeshGeoToolsLib::MeshNodeSearcher& searcher,
-	        const AssemblerLib::LocalToGlobalIndexMap& dof_table,
-	        const unsigned nodal_dof_idx,
-	        std::vector<GlobalIndexType>& global_ids, std::vector<double>& values);
+	template <typename OutputIterator>
+	void initializeDirichletBCs(
+	    OutputIterator bcs,
+	    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+	    const AssemblerLib::LocalToGlobalIndexMap& dof_table,
+	    const unsigned nodal_dof_idx)
+	{
+		for (auto& bc : _dirichlet_bcs)
+		{
+			DirichletBc<GlobalIndexType> new_bc;
+			bc->initialize(searcher, dof_table, nodal_dof_idx,
+			               new_bc.global_ids, new_bc.values);
+			bcs = new_bc;  // output iterator
```

Comment:
The disadvantages of the output iterator are:
* It is rather implicit (the assignment has to be commented for those not used to output iterators)
* Are there any performance issues because of resizing?